kv/kv.go

```diff
@@ -93,9 +95,19 @@ func NewLockCtx(forUpdateTS uint64, lockWaitTime int64, waitStartTime time.Time)
 }
 
 // InitReturnValues creates the map to store returned value.
-func (ctx *LockCtx) InitReturnValues(valueLen int) {
+func (ctx *LockCtx) InitReturnValues(capacity int) {
 	ctx.ReturnValues = true
-	ctx.Values = make(map[string]ReturnedValue, valueLen)
+	if ctx.Values == nil {
+		ctx.Values = make(map[string]ReturnedValue, capacity)
+	}
+}
+
+// InitCheckExistence creates the map to store whether each key exists or not.
+func (ctx *LockCtx) InitCheckExistence(capacity int) {
+	ctx.CheckExistence = true
```

Comment:
This field will be set to true only in tests?

Reply:
It's exported by TiDB (https://github.com/pingcap/tidb/blob/e7bd1faf5526921badda00ed96d98fd4c2ac2bf3/executor/executor.go#L1011)